src/spikeinterface/preprocessing/common_reference.py

```diff
@@ -1,25 +1,47 @@
+from __future__ import annotations
 import numpy as np
+from typing import Optional, Literal
 
 from spikeinterface.core.core_tools import define_function_from_class
 
 from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
 from ..core import get_closest_channels
+from spikeinterface.core.baserecording import BaseRecording
 
 from .filter import fix_dtype
 
 
 class CommonReferenceRecording(BasePreprocessor):
     """
-    Re-references the recording extractor traces.
+    Re-references the recording extractor traces. That is, the value of the traces are
+    shifted so the there is a new zero (reference).
+
+    The new reference can be estimated either by using common median reference (CMR) or
+    common average reference (CAR).
+
+    The new reference can be set three ways:
+        * "global": the median/average of all channels is set as the new reference.
+        That is, the median/average is subtracted from all channels.
+        * "single": In the simplest case, a single channel from the recording is set as the new reference.
+        That is, the channel is subtracted from all other channels. To use this option, the `ref_channel_ids` argument
+        is used with a single channel id. Note that this option will zero out the reference channel.
+        A collection of channels can also be used as the new reference. In this case, the median/average of the
+        selected channels is subtracted from all other channels. To use this option, pass the group of channels as
+        a list in `ref_channel_ids`.
+        * "local": the median/average within an annulus is set as the new reference. The parameters
+        of the annulus are specified using the `local_radius` argument. With this option, both close channels and
+        far channels are excluded from the median/average. Note that setting the `local_radius` to (0, exclude_radius)
+        will yield a simple circular local region.
+
```

Comment:
```suggestion
        * "global": the median/average of all channels is set as the new reference.
            In this case, the 'global' median/average is subtracted from all channels.
        * "single": In the simplest case, a single channel from the recording is set as the new reference.
            This channel is subtracted from all other channels. To use this option, the `ref_channel_ids` argument
            is used with a single channel id. Note that this option will zero out the reference channel.
            A collection of channels can also be used as the new reference. In this case, the median/average of the
            selected channels is subtracted from all other channels. To use this option, pass the group of channels as
            a list in `ref_channel_ids`.
        * "local": the median/average within an annulus is set as the new reference. 
                The parameters of the annulus are specified using the `local_radius` argument. With this option, both 
                channels which are too close and channels which are too far are excluded from the median/average. Note 
                that setting the `local_radius` to (0, exclude_radius)  will yield a simple circular local region.

```

A couple things: This was hard to read visually so I tried to add indents just to make it easier to read the simple explanation (after argument) and the detailed explanation (indented).

Also, (for English we don't typically use 'that is' that often. If you want to sound academic people will use 'ie' which is Latin for that is, but that is would only be used rarely. It's not wrong, it's just not standard.

Finally the local section still isn't quite clear. How can something be too close? (ie the relation between exclude_radius and local_radius is not completely clear just based on this docstring. You mention exclude_radius after you've mentioned that things too close will be exclude which is surprising for a reader (in my opinion).

Reply:
Thanks for the suggestions here. I will commit to check how the proposed visual style looks like as the diff is not a good vizualisation for this. Usually I go for this to read code (diff is bad for large PRs I feel):

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/99e9cdbe-f41a-49e0-86dd-f9cd32da52aa)

But here it does not work.

> Also, (for English we don't typically use 'that is' that often. 
Appreciate this, definitley a clutch of mine. 

As for the radius, I really think that the parameter is a little bit too general and is hard to explain. But basically, the exclude radius is used to exclude from the reference all the channels that are close to the current channel. So, for every channel you exclude channels that are too close and channels that are too far away. I kind of wish I could put a diagram of this in the notes:

![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/ba85855d-545c-4712-bbe0-ab6aa7549c58)

Any suggestions are welcomed.

Reply:
Is this what you intended?
![image](https://github.com/SpikeInterface/spikeinterface/assets/6429509/e386ee9f-d498-4dc5-96c7-2846fc88dafb)

My auto-corrector did a number on it so if what we have right now it is not feel free to push to my branch with your intended change.

Reply:
Yes that's what I wanted, except the "local" looks like it did a double set of spacing instead of single set. Tabbing wasn't working for me so I just did four spaces for each section. Not sure if it was your autocorrect or me mistyping.

Reply:
Should be corrected now.